pkg/products/threescale/reconciler.go

```diff
@@ -1551,3 +1575,18 @@ func (r *Reconciler) deleteConsoleLink(ctx context.Context, serverClient k8sclie
 	}
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
+
+func (r *Reconciler) reconcileDeploymentConfigs(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+
+	ns := global.NamespacePrefix + defaultInstallationNamespace
+
+	for _, name := range threeScaleDeploymentConfigs {
+		deploymentConfig := &appsv1.DeploymentConfig{}
+
+		_, err := resources.ReconcileZoneTopologySpreadConstraints(ctx, serverClient, k8sclient.ObjectKey{Name: name, Namespace: ns}, resources.SelectFromDeploymentConfig, "app", deploymentConfig)
+		if err != nil {
+			return integreatlyv1alpha1.PhaseInProgress, err
+		}
+	}
```

Comment:
@CathalOConnorRH , I think you might need an "PhaseAwaitingComponents" check here

Reply:
Thanks Brian. I've updated now.